### DISCUSSION TITLE
Add external local storage after creating instance

### DISCUSSION THREAD

**Discussion by aristotelis-economides** [nextcloud/all-in-one#1353] [Questions] [answered]
Hi! 

First of all, thank you for the amazing ongoing work with the Nextcloud AIO image!

I have my instance already running, and I want to add an external SSD as external local storage. From the [readme](https://github.com/nextcloud/all-in-one#how-to-allow-the-nextcloud-container-to-access-directories-on-the-host), I can understand that one can do this when creating the mastercontainer.

It's not clear to me whether I should delete the mastercontainer (and therefore lose my current data/setup), to be able to do this? Or is there a way to restart the container with the `-e NEXTCLOUD_MOUNT= ...` flag enabled? 

I'm sorry if this is a newbie question, I am not that well versed in Docker, and this might be quite a trivial question.

Thanks!

**Comment by Zoey2936** [comment, marked as answer]
you will only lose your data if you delete the mastercontainer volume

**Reply by aristotelis-economides** [reply]
Okay I see. Then, if I understand correctly I need to stop all the containers, and then start the mastercontainer again (using `docker start  [OPTIONS] CONTAINER [CONTAINER...]` right? Is there a particular order or way that the containers should be stopped or is `docker stop ...` enough?

**Reply by Zoey2936** [reply]
You just need to stop the mastercontainer:
`docker stop nextcloud-aio-mastercontainer`
Remove it (removing the container != removing the volume):
`docker rm nextcloud-aio-mastercontainer`

And then use the initial docker run command from the readme with your options set using env again.
Then just open the aio interface, stop there all containers and restart them

**Reply by aristotelis-economides** [reply]
All right!! Thank you so much! :)

**Reply by Blandness4047** [reply]
can you please explain how to restart only the master container. should I have to follow the initial steps to setup Nextcloud? if so what will happen to other running containers?

**Reply by aristotelis-economides** [reply]
Hey, I'm pretty basic with my Docker understanding, but IIUC, if you run the command from the readme: 

```bash
sudo docker run \
--sig-proxy=false \
--name nextcloud-aio-mastercontainer \
--restart always \
--publish 80:80 \
--publish 8080:8080 \
--publish 8443:8443 \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
nextcloud/all-in-one:latest
```

This only starts the mastercontainer. Which is then responsible for spinning up/restarting the rest of the containers. Also, if you don't tamper with the storage volume itself, which per the command is stored in `/mnt/...`, then your data won't be affected.

**Reply by Blandness4047** [reply]
Thanks. It worked
